hypershift-operator/controllers/hostedcluster/hostedcluster_webhook.go

```diff
@@ -10,23 +10,31 @@ import (
 
 	corev1 "k8s.io/api/core/v1"
 
+	configv1 "github.com/openshift/api/config/v1"
 	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
 	"k8s.io/apimachinery/pkg/api/equality"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
 )
 
 // Webhook implements a validating webhook for HostedCluster.
-type Webhook struct{}
+type Webhook struct {
+	client.Client
+}
 
 // SetupWebhookWithManager sets up HostedCluster webhooks.
 func SetupWebhookWithManager(mgr ctrl.Manager) error {
+	var webhook Webhook
+	webhook.Client = mgr.GetClient()
```

Comment:
In order to not start a cache for networks and just fetch it once during startup, lets use the uncached client:
```suggestion
	webhook.Client = mgr.GetAPIReader()
```